napari/components/viewer_model.py

```diff
@@ -904,11 +913,119 @@ def open(
             else None,  # indeterminate bar for 1 file
         ) as pbr:
             for _path in pbr:
+                # TODO: where can I put tests for this so that I have access to different plugins to try
+                if plugin:
+                    added.extend(
+                        self._add_layers_with_plugins(
+                            _path, kwargs, plugin=plugin, layer_type=layer_type
+                        )
+                    )
+                # no plugin choice was made
+                else:
+                    readers = get_potential_readers(_path)
+                    if select_reader_helper is None:
```

Comment:
so we are going to go with the UX that someone using `viewer.open` in, for example, the builtin console of a gui-based session gets an exception?  I can see the merits, but it's also a relatively consequential decision, pinging @jni for his thoughts here.  (question is, should a direct call to `viewer.open` have any context awareness, or _always_ raise an exception if two plugins are capable of doing the reading)

Reply:
This is one way in which the "call-order" preference allowed the API user to benefit from multiple plugins without involving a widget.  So, perhaps in parallel to this, we could consider ways in which preferences would help us to avoid needing to raise an exception on what might be a relatively common use case